setup.sh

```diff
@@ -14,29 +14,34 @@ normalize_version() {
 }
 
 latest_version() {
-  jq --arg channel "$1" '.releases | map(select(.channel==$channel)) | first'
+  jq --arg channel "$1" --arg dart_sdk_arch "$ARCHITECTURE" '.releases | map(select(.channel==$channel) | select(.dart_sdk_arch==$dart_sdk_arch)) | first'
 }
 
 wildcard_version() {
-  if [[ $1 == any ]]; then
-    jq --arg version "^$2" '.releases | map(select(.version | test($version))) | first'
+  if [[ $2 == *"v"* ]] || [[ $2 == 1 ]] || [[ $2 == 0 ]]; then  # is legacy version format
```

Comment:
Using the above approach, I think we can simplify it. Having fewer conditionals is always better.